### ISSUE TITLE
Get the parent and siblings of child XCUIElement 

### ISSUE THREAD

**Issue by himankarbn** [appium/ruby_console#78]
# Example:
## Screenshot
![screen shot 2017-09-07 at 1 38 44 pm](https://user-images.githubusercontent.com/11064961/30179424-f6b74928-93d1-11e7-9a0c-6760513021fd.png)

## XML
![screen shot 2017-09-07 at 1 37 31 pm](https://user-images.githubusercontent.com/11064961/30179454-11cf2a00-93d2-11e7-9ee4-4ffdc32c1a77.png)

### Using XCTest I can find the sibling using child StaticText containing `Account:` using the below query:

`app.table.cells.containing(NSPredicate(format: " label CONTAINS[c] 'Account:' ")).children(matching: .textField).element`

#### How can I find the parent cells (XCUIElementCell) and siblings using child (XCUIElementStaticText containing Account: text ) in appium ruby?

Basically traverse the XML tree upwards without using XPATH.

**Comment by himankarbn** [comment]
@mykola-mokhnach^^ @KazuCocoa^^ help please

**Comment by mykola-mokhnach** [comment]
I think it's impossible for now without xpath usage. There is a pending [PR](https://github.com/facebook/WebDriverAgent/pull/707) in WebDriverAgents, which adds _containing_ support into class chain requests.

**Comment by himankarbn** [comment]
@mykola-mokhnach : that will be awesome. thanks!

**Comment by KazuCocoa** [comment]
Or, ruby client provide `textfields` methods to get all `XCUIElementTypeTextField` and `XCUIElementTypeSecureTextField` on the page. So, if the cell's position doesn't change dynamically, you also can get the text field with `textfirlds[0]`, the `0` should be changed to fit your case.
@himankarbn

**Comment by himankarbn** [comment]
@KazuCocoa : thanks for the helpful tip but I was looking something similar to `containing(NSPredicate(format: " label CONTAINS[c] '<string>' ")) is available in the Apples's new XCTest or the `REVERSE XPATH` . Most of the fields in our app is dynamic so we are looking for ways to get the parent cell and following siblings cells based on name/text/value of a cell.
The above was just an example. 

As of now we are using XPath but excited waiting for the [PR](https://github.com/facebook/WebDriverAgent/pull/707) to get merged.
